backend/models/application_details.py

```diff
@@ -0,0 +1,64 @@
+from pydantic import BaseModel
+
+from typing import Dict
+
+from backend.models.academics.section import Section
+from backend.models.application import Application, UTA, New_UTA, Returning_UTA
+
+from backend.models.user import User
+
+__authors__ = ["Ben Goulet"]
+__copyright__ = "Copyright 2024"
+__license__ = "MIT"
+
+
+class ApplicationDetails(Application):
+    """
+    Pydantic model to represent an `Application`.
+
+    This model is based on the `ApplicationEntity` model, which defines the shape
+    of the `Application` database in the PostgreSQL database.
+    """
+
+    user: User
+
+
+class UTADetails(UTA, ApplicationDetails):
+    """
+    Pydantic model to represent a `UTA Application`.
+
+    This model is based on the `UTA` model, which defines the shape
+    of the `Application` database in the PostgreSQL database.
+    """
+
+    preferred_sections: list[Section]
+
+
+class New_UTADetails(New_UTA, UTADetails):
+    """
+    Pydantic model to represent a `UTA Application`.
+
+    This model is based on the `UTA` model, which defines the shape
+    of the `Application` database in the PostgreSQL database.
+    """
+
+
+class Returning_UTADetails(Returning_UTA):
+    """
+    Pydantic model to represent a `UTA Application`.
+
+    This model is based on the `UTA` model, which defines the shape
+    of the `Application` database in the PostgreSQL database.
+    """
+
+
+class UserApplication(BaseModel):
```

Comment:
Wondering if this model is redundant if it is just a wrapper for `New_UTADetails`! Also, should this return new details always, even when students are returning applicants?

Reply:
Yes it is redundant! That is my fault, I had it set up like this just in case I had to approach ordinality issues in a different manner and return 'preferred sections' in the model with the application. Will remove though.